x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -66,9 +70,24 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
 
   const kibana = useMlKibana();
   const {
-    services: { share, application },
+    services: { data, share, application },
   } = kibana;
 
+  const getMapsLink = async (anomaly: AnomaliesTableRecord) => {
+    const initialLayers = getInitialAnomaliesLayers(anomaly.jobId);
+    const anomalyBucketStart = moment(anomaly.time).toISOString();
+    const timeRange = data.query.timefilter.timefilter.getTime();
+    // Set 'from' in timeRange to start bucket time for the specific anomaly
+    timeRange.from = anomalyBucketStart;
```

Comment:
You should also set `timeRange.to` to be the end of the bucket (minus 1ms), and add a filter for the partitioning field(s) as you do when opening Maps from the embedded map.

Reply:
Added timerange and query for partitioning fields in 1a4013bf62e01b1d3f5cde8fd6454843de4ccb2e